components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -931,6 +932,7 @@ public static X509CredentialImpl getX509CredentialImplForTenant(String tenantDom
         KeyStore keyStore;
 
         try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
```

Comment:
tenant flow needs to be started and ended only for the tenant case. Can we move the tenant flow starting logic within the
``if (tenantId != -1234) {``` block

A similar fix is done in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1412/files

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/307/commits/5935fb2cd0d6b08ffb6efd20003eaa1e0aff90b5